src/encode/pattern/mod.rs

```diff
@@ -584,22 +612,24 @@ impl FormattedChunk {
             FormattedChunk::Newline => w.write_all(NEWLINE.as_bytes()),
             FormattedChunk::Align(ref chunks) => {
                 for chunk in chunks {
-                    chunk.encode(w, record)?;
+                    chunk.encode(w, record, color_map)?;
                 }
                 Ok(())
             }
             FormattedChunk::Highlight(ref chunks) => {
-                match record.level() {
-                    Level::Error => {
-                        w.set_style(Style::new().text(Color::Red).intense(true))?;
+                if let Some(Some(color)) = color_map.get(&record.level()) {
```

Comment:
What happens when we get None? or Some(None)?

Reply:
Then the text is not styled and output as regulate unstyled text.